### DISCUSSION TITLE
For the sake of 12pixels ~ deliberate?

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1085] [Q&A]
Odd title, more of my OCD when using GUIs, but what I'm referencing here is this ;

![GUI area under my review](https://i.imgur.com/5i7dNEs.png)

I will do that, for the sake of selecting the (still visible) bottom most button with a single mouse click, as opposed to clicking on the scrollbar down button..to make the still visible button more visible (readable), before using a 2nd mouse click to achieve the same ends.  If that sounds redundant, then perhaps it is, but it's still an example of interrupting the GUI workflow in usage, 'needlessly' ....ie; the scrollbar has to justify it's existence ~ by rights it should totally hide the bottom button(s), so it's actually needed.

The reason I append 'deliberate?' is because I see what happens if GUI_HEIGHT is set to 612 ~ scroll area/bar are in that case not needed, and are replaced by vacant white area (SHOW_ALWAYS for the column area iirc), and it looks very ugly... and at a glance what should be the actual case there is no scrollbar, and just the panel divider (refactored to the left perhaps)....but this is not an error or fault in any way, the GUI has been written like this ~ I'm perhaps interested in knowing why it's like this? ..ie; something to do with SDL perhaps?

**Comment by midwan** [comment]
It's not deliberate, it just happened to be this way when the new option was added.
We could increase the spacing on all items a bit, and that would push it down further and hide it, although ideally I'd prefer not to have any scrolling needed there.

I also tried changing the font size, that that would help in squeezing everything in as well, but the specific font used looks ugly in any smaller size than the current one.

**Comment by giantclambake** [comment]
Hmm, okay...so not by design, and like myself you're thinking a design involving no scrollbar would be the preferred design.

Agreed wrt the font ; I meant to ask if an Amiga TTF was being used for any particular reason.

Regarding that panel & with a view to make it all 'fit' with no need of scrollbar, my brain wandered off wondering if the functions of one button item, could be incorporated into another...thus removing item count & need for scrollbar, and actually enlarging each item area to make fit. If it came to thinking about that ... combining RAM & ROM makes a good deal of sense, and is an easy fit for the right panel...?

**Comment by giantclambake** [comment]
Just mocking this up in the gimp, combining the RAM and ROM panels on a single button would look like this....

![mock-up of RAM and ROM panels](https://i.imgur.com/KhX2Nl4.png)

The single pro you get with that, is it requires one less button, which then removes the need for the scrollbar... however, it follows on. The area recovered by no scrollbar, might be refilled by refactoring the button size/font .. or you leave the buttons alone, and just resize main menu panel area accordingly (move divider?). If that were the easier solution, any vertical space created by removing one button, I'd pad by enlarging the 'About' button, under the auspices of subtle self-promotion.  B^)

Ofc, one has to question, would this confuse the amiberry user, combining ram & rom selection/settings into 1 widget? I don't think so... if you read in that real amiga kickstart roms are in place as they should be, then ram & rom selections are largely dictated by user selecting Amiga machine type, and for the 'casual' user this is largely automated. If you're not a casual user, you'll see this panel and won't be confused...(although looking at it now, ramfield selector should be at the top, and rom at the bottom...)

There is another  button besides RAM & ROM that is worth discussing here -- 'IO Ports' ... if not only due to 'Input' & 'IO Ports'  being so ambiguously disposed sitting beside each other =) 

For a start, IO Ports --> Sampler:   ...this is unmentioned on that panel's helptext, and that's likely a good thing because IMHO this belongs in the 'Sound' panel settings (it'd probably fit in under the 'floppy drive sound emulation' & 'sound buffer size' boxes if you moved those up a little and trimmed a bit of height?)....but my brain?...looking for sampler settings, I'd check 'Sound' first =)

Technically speaking, as this sampler switch only hooks audio capture interface(s), it is really a 'capture only' sampler, input device...(not a combined capture/soundcard device, which is/would be IO). 

...IO Ports --> Serial port:   ....this is a difficult one to place context upon, akin to asking 'why isn't the parallel port here?' Ignore my blather here, it maybe desktop computer usage bound... or maybe a question -- what do ppl use this serial<=>usb translation for? I mean, from the Amiga machine viewpoint, this presents itself mostly as a USB (serial) addon card if it points to /dev/ttyUSB0 , whereas it's just an interface to /dev/ttyS0 on machines with UART serial port (many thin clients do), and in this case you're just replicating the serial port on Amiga hardware ...so like I say, a difficult thing to contextualize ~ it depends on port standard used.

If I was trying to convince myself to put serial port somewhere else, I'd likely park it in 'Expansions' ...actually, 'Expansions -> Miscellaneous Expansions' would perhaps be a fitting place for it, just because the other uae**.devices expose themselves in that location....

...IO Ports --> Protection Dongle:  .... dongle is a thing (endpoint), not a port ; in the real world dongles plugin to IO ports...ie; they're devices, not ports. What's really happening, is a synthesis of both dongle & port, in an 'invisible bridge' way. If we were on Amiga, we'd consider this thing to be a required hardware 'accessory'? ...like, it's more a necessary part of the software title itself, than it is anything else ....but it's not an IO Port..... of this, I am certain 8)  In a sense, wrt the emulated guest OS, you're actually replicating both the IO port and the dongle itself ....

...and I can convince myself the protection dongle widget is similarly disposed to the serial port widget, and could be considered to be a Miscellaneous Expansions candidate, and this'd all fit with a bit of moving about in the Expansions panel code.

Hypothetically, this would free-up another button location ~ what you did with that would be up for grabs ... but for example, if you wanted to expose GUI customization/themes widgets somewhere in the GUI itself, coupled with the fact there's a few items in 'Miscellaneous' I could deem to be directly GUI related (and not so much emulator control), you could group all of these behind their own button/panel of  'GUI Prefs' or such'n'similar.....ie; in context the 'start in quickstart mode' is a GUI prefs thing really =) I do perhaps appreciate why it is where it is...but then amiberry GUI never had a GUI prefs panel, so the thought chases it's tail.

Still...the moment you say no scrollbar, the above is what immediately springs to mind in my olde aged head  B)

**Reply by midwan** [reply]
The GUI design is meant to replicate that of WinUAE as closely as possible, so a lot of the design choices and items placement is coming from there. We can of course make modifications where it makes sense or where it's relevant (e.g. there is no Parallel options area, because Parallel port support is not implemented yet), but I'm trying to keep it consistent with the look and feel of WinUAE.

**Comment by giantclambake** [comment]
Ahh, I didn't appreciate you were trying to follow the winuae example ~ I had to install winuae here under WINE just to have a look at it's GUI (because....) I don't use windows and have never used winuae....ie; the 'look and feel' of winuae is a total unknown to me.

Having done that, I can appreciate the similarities in 'look', but when considering GUI 'feel' ....nah, perhaps this is at the crux of much of my discussion here ; now that I've had a little play with winuae, perhaps I would say it like this...

"winuae is every bit the desktop app that amiberry is not, whereas amiberry is every bit the small device GUI app that winuae is not"

For example...visual appreciation ..in WINE at least, winuae opens a 1280x1024 GUI window, there is no 'suggestion' that one might want to combine the RAM & ROM entries, because the contents of the panels are scaled/spaced so as to present no free/vacant area, and there's no incentive to do that due to GUI size and presentation.

Conversely... right.... as the amiberry GUI is primarily aimed at catering for small devices/800x600 window, there is imho underlying incentive to make practical use of all GUI areas/fabric as much as is possible, and combining RAM & ROM would achieve that end, in keeping with the winuae 'look', but having an amiberry GUI 'feel', consummate with small device targets... ie; I doubt winuae would cater for small devices, as well as amiberry current presents itself as a desktop app...

...otherwise, I end up with a synopsis that the RAM and ROM panel contents in amiberry be spaced out accordingly, to more closely approximate the look & feel of the winuae GUI   ...which is something I would NOT suggest, because then the panel contents look sparse/stoopid  (imho =) and such contradicts the underlying incentive of trying to logically/functionally present something in a 'condensed' GUI ....you sometimes see this with desktop apps ported the mobilephones etc, where a significant rework of GUI menus and settings etc take place. 

I think this is why my first surmise sometime previously, was that the 'sanest' way to do any of this, would be at compile time ; the current amiberry GUI build target, and a desktop amiberry GUI build target. As I'm more than 6 decades old, I do appreciate the amount of work that would entail (when I did code, it was with BCPL on industrial process control 'frames =) That is compounded by your previous mention of using Qt gui toolkit, and perhaps that could make for the ends of a means to obtain a better 'desktop application' GUI build of amiberry, whilst retaining the lightweight and condensed SDL based GUI for small devices....(but really, SDL is fine at present)....

...I know this might sound conceptual and/or subjective, but the reason I say SDL is fine is because it is that flexible ~ if I start amiberry now, I'm presented with a configuration GUI --- the moment you change gui height/width to say 1024x768, and amiberry starts up with that sized GUI, it presents itself more as a desktop app (like winuae does =), and more than a config GUI.... even though it's still a config GUI ~ the fact that it's bigger/more substantial, is enough to push it into the desktop app realm.

The moment you do that of course, you have heaps or additional GUI area/canvas to play with, you don't need a scrollbar, and you almost want more buttons just because there now room to think like that. I do appreciate this 'could just be me', or perhaps it's just because I'd like to see amiberry have the same 'look & feel' of a desktop app, as fs-uae-launcher or winuae... something like that =)

I might see if I can learn enough about SDL, and make sense of a 1024x768 GUI build ...moving/resizing GUI elements is (hopefully) not out of my comfort zone (yet)...hint: the older one gets, the harder it becomes to learn new tricks 8)

As you mention parallel port support... query: for what purpose/use is that employed wrt Amiga emulation?

**Comment by giantclambake** [comment]
I suppose another way to look at this, is from a functional semantics viewpoint...

This is to say,  the 'About' entry in left-pane, is NOT a configuration item ~ the other 22 items ARE configuration items.

As there is room to do so, the 'About' entry could be moved from the left pane, down to the GUI main window buttons area (shutdown, quit, restart, help, reset, start.. +about)

If then the  'Start in quickstart mode' widget is unset, you could display the About panel by default when GUI starts, and users have to select one of the configuration items in the left column to display the corresponding config panel for that item. 

If the  'Start in quickstart mode' widget is set, by default the Quickstart panel is displayed.

That way we can remove the left-pane scrollbar, with minimal impact wrt the look&feel of winuae, and retain the 'About' functionality?

**Comment by giantclambake** [comment]
Fixed by GUI enhancements in preview 6.1.x =)
